File: articles/the-history-of-futures-thinking/fortran/history_score.f90
program history_score
  implicit none
  real :: methodological, participation, ethics, systems, institutional_power
  real :: reflective_score, power_risk

  methodological = 0.68
  participation = 0.92
  ethics = 0.86
  systems = 0.64
  institutional_power = 0.42

  reflective_score = 0.25*methodological + 0.25*participation + 0.25*ethics + 0.25*systems
  power_risk = institutional_power * (1 - participation) * (1 - ethics)

  print *, "Participatory and Democratic Futures reflective score:", reflective_score
  print *, "Participatory and Democratic Futures power risk:", power_risk
end program history_score

File: articles/the-history-of-futures-thinking/c/history_score.c
#include <stdio.h>

double reflective_score(double methodological, double participation, double ethics, double systems) {
    return 0.25 * methodological + 0.25 * participation + 0.25 * ethics + 0.25 * systems;
}

double power_risk(double institutional_power, double participation, double ethics) {
    return institutional_power * (1.0 - participation) * (1.0 - ethics);
}

int main(void) {
    double score = reflective_score(0.68, 0.92, 0.86, 0.64);
    double risk = power_risk(0.42, 0.92, 0.86);
    printf("Participatory and Democratic Futures: reflective_score=%.4f power_risk=%.4f\n", score, risk);
    return 0;
}
